aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,994 @@
+            });
+            i = i + 1;
+        };
+    }
+
+    /// Mint source certificate.
+    public entry fun mint_source_certificate(
+        nft_claimer: &signer,
+        amount: u64
+    ) acquires NFTMintConfig, PublicMintConfig, WhitelistMintConfig, SourceToken, CollectionConfig {
+        assert!(exists<CollectionConfig>(@post_mint_reveal_nft) && exists<WhitelistMintConfig>(@post_mint_reveal_nft) && exists<PublicMintConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+
+        let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(@post_mint_reveal_nft);
+        let public_mint_config = borrow_global_mut<PublicMintConfig>(@post_mint_reveal_nft);
+
+        let now = timestamp::now_seconds();
+        let is_whitelist_minting_time = now > whitelist_mint_config.whitelist_minting_start_time && now < whitelist_mint_config.whitelist_minting_end_time;
+        let is_public_minting_time = now > public_mint_config.public_minting_start_time && now < public_mint_config.public_minting_end_time;
+        assert!(is_whitelist_minting_time || is_public_minting_time, error::permission_denied(EMINTING_IS_NOT_ENABLED));
+        let claimer_addr = signer::address_of(nft_claimer);
+
+        // if this is the whitelist minting time
+        if (is_whitelist_minting_time) {
+            assert!(bucket_table::contains(&whitelist_mint_config.whitelisted_address, &claimer_addr), error::permission_denied(EACCOUNT_NOT_WHITELISTED));
+            let remaining_mint_allowed = bucket_table::borrow_mut(&mut whitelist_mint_config.whitelisted_address, claimer_addr);
+            assert!(*remaining_mint_allowed >= amount, error::invalid_argument(EAMOUNT_EXCEEDS_MINTS_ALLOWED));
+            while (amount > 0) {
+                mint_source_certificate_internal(nft_claimer, whitelist_mint_config.whitelist_mint_price);
+                amount = amount - 1;
+                *remaining_mint_allowed = *remaining_mint_allowed - 1;
+            };
+        } else {
+            let collection_config = borrow_global<CollectionConfig>(@post_mint_reveal_nft);
+            if (option::is_some(&collection_config.public_mint_limit)) {
+                // If the claimer's address is not on the public_minting_addresses table yet, it means this is the
+                // first time that this claimer mints. We will add the claimer's address and remaining amount of mints
+                // to the public_minting_addresses table.
+                if (!bucket_table::contains(&public_mint_config.public_minting_addresses, &claimer_addr)) {
+                    bucket_table::add(&mut public_mint_config.public_minting_addresses, claimer_addr, *option::borrow(&collection_config.public_mint_limit));
+                };
+                let limit = bucket_table::borrow_mut(&mut public_mint_config.public_minting_addresses, claimer_addr);
+                assert!(amount <= *limit, error::invalid_argument(EAMOUNT_EXCEEDS_MINTS_ALLOWED));
```

Comment:
Can there be a use case where public mint has no per-account limit?

Reply:
yeah if the collection is not untracked or not popular, the creators might have concerns over the collection not being minted out